### PR TITLE
Support branch "create" action

### PR DESCRIPTION
**Pull request by quantonganh**
https://github.com/drone/drone/issues/2685

Discussion: https://gitter.im/drone/drone?at=5ff5c66acd31da3d9a961645

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -75,3 +75,5 @@ require (
 replace github.com/h2non/gock => gopkg.in/h2non/gock.v1 v1.0.14
 
 go 1.13
+
+replace github.com/drone/go-scm => github.com/quantonganh/go-scm v1.8.1-0.20210107022054-cb9eef77a93f
```
Comment (techknowlogick) [inline go.mod:79]: This should be removed prior to merging (but after blocking PR is merged)

Comment (quantonganh) [reply go.mod:79]: Of course. It's temporary to make the test passed.

**Comment by quantonganh** [comment]
As @bradrydzewski [pointed out](https://github.com/drone/go-scm/pull/92#issuecomment-755852095), no need to add a new field to know if a branch has been created. We can do that by checking if the before commit is empty. Updated my PR.

**Comment by tphoney** [comment]
Hi @quantonganh i am unsure of this newly added behaviour, as it changes what existing drone users would expect. To move this forward in an open way i would suggest adding a proposal here https://github.com/drone/proposal. So we can design with the rest of the community what the behaviour should be and if the community as a whole needs this functionality.

For now i am going to close this PR. Thank you for your submission and the time and effort you have put in.

**Pull request closed without merging**